node/pkg/chain/helper/helper.go

```diff
@@ -269,15 +251,41 @@ func (t *ChainHelper) PublicAddressString() (string, error) {
 }
 
 func (t *ChainHelper) retryOnJsonRpcFailure(ctx context.Context, job func(c utils.ClientInterface) error) error {
+	var err error
 	for _, client := range t.clients {
-		err := job(client)
+		err = job(client)
 		if err != nil {
 			if utils.ShouldRetryWithSwitchedJsonRPC(err) {
 				continue
 			}
 			return err
 		}
+		return nil
+	}
+	return err
+}
```

Comment:
when there is a json rpc failure, can it include the transaction being picked up from mempool and processed? If so, that'll increment the wallet's transaction count and we need to `getAndIncrement` new nonce for every json rpc retry. If json rpc fail doesn't include a transaction processing from mempool then it shouldn't be a problem

Reply:
I've made an update, now it'll use a same nonce within json rpc failure